packages/fluent-ui/src/WrapIfAdditionalTemplate/WrapIfAdditionalTemplate.tsx

```diff
@@ -1,11 +1,75 @@
-import { FormContextType, RJSFSchema, StrictRJSFSchema, WrapIfAdditionalTemplateProps } from '@rjsf/utils';
+import { FocusEvent } from 'react';
+import { TextField } from '@fluentui/react';
+import {
+  ADDITIONAL_PROPERTY_FLAG,
+  FormContextType,
+  RJSFSchema,
+  StrictRJSFSchema,
+  TranslatableString,
+  WrapIfAdditionalTemplateProps,
+} from '@rjsf/utils';
 
 export default function WrapIfAdditionalTemplate<
   T = any,
   S extends StrictRJSFSchema = RJSFSchema,
   F extends FormContextType = any
 >(props: WrapIfAdditionalTemplateProps<T, S, F>) {
-  const { children } = props;
-  // TODO Implement WrapIfAdditionalTemplate features in FluentUI (#2777)
-  return <>{children}</>;
+  const {
+    children,
+    classNames,
+    disabled,
+    id,
+    label,
+    onDropPropertyClick,
+    onKeyChange,
+    readonly,
+    registry,
+    required,
+    schema,
+    style,
+    uiSchema,
+  } = props;
+  const { templates, translateString } = registry;
+  // Button templates are not overridden in the uiSchema
+  const { RemoveButton } = templates.ButtonTemplates;
+  const keyLabel = translateString(TranslatableString.KeyLabel, [label]);
+  const additional = ADDITIONAL_PROPERTY_FLAG in schema;
+
+  if (!additional) {
+    return (
+      <div className={classNames} style={style}>
```

Comment:
Most of the snapshot changes come from this additional `div`. This was added so that the `classNames` and `style` properties have an effect. This matches what is done in all other themes.

Reply:
@tjasz Can you replace this with the `div` from the `FieldTemplate` and remove it from there? This will make `fluent-ui` consistent with the others. Also you may want to either pass in the `classNames` defined in `FieldTemplate` to `WrapIfAdditionalTemplate` or move that code here as well.

Reply:
Thanks for pointing me to that. That was the right thing to do. The tests look a lot better. The Array and Form tests still have some diffs where `display: undefined` was removed, but no more extra `<div>`s.